backend/src/graphql/resolver/AdminResolver.test.ts

```diff
@@ -1324,4 +1328,252 @@ describe('AdminResolver', () => {
+          )
+          for (let i = 0; i < bibisTransaktionLinks.length; i++) {
+            await transactionLinkFactory(testEnv, bibisTransaktionLinks[i])
+          }
+
+          // admin: only now log in
+          await query({
+            query: login,
+            variables: { email: 'peter@lustig.de', password: 'Aa12345_' },
+          })
+        })
+
+        afterAll(async () => {
+          await cleanDB()
+          resetToken()
+        })
+
+        describe('without any filters', () => {
+          it('finds 6 open transaction links and no deleted or redeemed', async () => {
+            await expect(
+              query({
+                query: listTransactionLinksAdmin,
+                variables,
+              }),
+            ).resolves.toEqual(expectNoDeletedOrRedeemed)
+          })
+        })
+
+        describe('all filters are null', () => {
+          it('finds 6 open transaction links and no deleted or redeemed', async () => {
+            await expect(
+              query({
+                query: listTransactionLinksAdmin,
+                variables: {
+                  ...variables,
+                  filters: {
+                    withDeleted: null,
+                    withExpired: null,
+                    withRedeemed: null,
+                  },
+                },
+              }),
+            ).resolves.toEqual(expectNoDeletedOrRedeemed)
+          })
+        })
+
+        describe('filter with deleted', () => {
+          it('finds 6 open transaction links and 1 deleted and no redeemed', async () => {
+            await expect(
+              query({
+                query: listTransactionLinksAdmin,
+                variables: {
+                  ...variables,
+                  filters: {
+                    withDeleted: true,
+                    withExpired: null,
+                    withRedeemed: null,
+                  },
+                },
+              }),
+            ).resolves.toEqual(
+              expect.objectContaining({
+                data: {
+                  listTransactionLinksAdmin: {
+                    linkCount: 7,
+                    linkList: expect.arrayContaining([
+                      expect.not.objectContaining({
+                        memo: 'Leider wollte niemand meine Gradidos zum Neujahr haben :(',
+                        createdAt: expect.any(String),
+                      }),
+                      expect.objectContaining({
+                        memo: 'Da habe ich mich wohl etwas übernommen.',
+                        deletedAt: expect.any(String),
+                      }),
+                    ]),
+                  },
+                },
+              }),
+            )
+          })
+        })
+
+        // TODO: works not as expected, because the expired link is not in the result eventhogh a link has 'createdAt' at the very first of 2022 and should be expired
+        describe.skip('filter by expired', () => {
```

Comment:
@Mogge if you like to have it done yet, we should do it together

Reply:
This one should work. See my suggestions below.

Reply:
Thanks for your help and hint with the pagination count! @Mogge